main/tasks.py

```diff
@@ -64,3 +67,88 @@ def daily_project_status_check() -> None:
     projects = Project.objects.filter(status="Active")
     for project in projects:
         project.check_and_notify_status()
+
+
+_template_time_logged = """
+Dear {project_leader},
+
+This is your monthly summary of project work. In {last_month_name} you have logged:
+
+{project_work_summary}
+
+You have invested on project work {percentage}% of your time.
+
+If you have more time to log for {last_month_name}, please do so by the 10th of
+{current_month_name} in [Clockify](https://clockify.me/).
+
+Best wishes,
+ProCAT
+"""
+
+
+# Runs on the 3rd day of every month at 10:00 AM
+@db_periodic_task(crontab(day=3, hour=10))
+def notify_monthly_time_logged_summary() -> None:
+    """Notify users about their monthly time logged."""
+    from .models import TimeEntry, User
+
+    avg_work_days_per_month = 220 / 12  # Approximately 18.33 days per month
+
+    today = datetime.today()
+
+    last_month_start = datetime(year=today.year, month=today.month - 1, day=1)
+    current_month_start = datetime(year=today.year, month=today.month, day=1)
+
+    time_entries = TimeEntry.objects.filter(
+        start_time__gte=last_month_start, end_time__lt=current_month_start
+    )
+
+    if not time_entries.exists():
+        return  # No entries to process
+
+    users = User.objects.filter(timeentry__in=time_entries).distinct()
+
+    for user in users:
+        user_entries = time_entries.filter(user=user)
+
+        project_hours: dict[str, float] = {}
```

Comment:
You can set the default value directly here using the [defaultdict](https://docs.python.org/3/library/collections.html#collections.defaultdict).

```python
from collections import defaultdict

project_hours = defaultdict(float) # <- This defaults to 0.0
```